src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.OneShot.cs

```diff
@@ -0,0 +1,300 @@
+            int outputLength)
+        {
+            if (outputLength < 0)
+                throw new ArgumentOutOfRangeException(nameof(outputLength), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            byte[] result = new byte[outputLength];
+            Pbkdf2(password, salt, result, iterations, hashAlgorithm);
+            return result;
+        }
+
+        /// <summary>
+        /// Fills a buffer with a PBKDF2 derived key.
+        /// </summary>
+        /// <param name="password">The password used to derive the key.</param>
+        /// <param name="salt">The key salt used to derive the key.</param>
+        /// <param name="iterations">The number of iterations for the operation.</param>
+        /// <param name="hashAlgorithm">The hash algorithm to use to derive the key.</param>
+        /// <param name="destination">The buffer to fill with a derived key.</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="iterations" /> is not a positive value.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="hashAlgorithm" /> has a <see cref="HashAlgorithmName.Name" />
+        ///   that is empty or <see langword="null" />.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <paramref name="hashAlgorithm" /> is an unsupported hash algorithm. Supported algorithms
+        ///   are <see cref="HashAlgorithmName.SHA1" />, <see cref="HashAlgorithmName.SHA256" />,
+        ///   <see cref="HashAlgorithmName.SHA384" />, and <see cref="HashAlgorithmName.SHA512" />.
+        /// </exception>
+        /// <exception cref="EncoderFallbackException">
+        /// <paramref name="password" /> contains text that cannot be converted to UTF8.
+        /// </exception>
+        /// <remarks>
+        /// The <paramref name="password" /> will be converted to bytes using the UTF8 encoding. For
+        /// other encodings, convert the password string to bytes using the appropriate <see cref="System.Text.Encoding" />
+        /// and use <see cref="Pbkdf2(ReadOnlySpan{byte}, ReadOnlySpan{byte}, Span{byte}, int, HashAlgorithmName)" />.
+        /// </remarks>
+        public static void Pbkdf2(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            Span<byte> destination,
+            int iterations,
+            HashAlgorithmName hashAlgorithm)
+        {
+            const int MaxPasswordStackSize = 256;
+
+            byte[]? rentedPasswordBuffer = null;
+            int maxEncodedSize = s_throwingUtf8Encoding.GetMaxByteCount(password.Length);
+
+            Span<byte> passwordBuffer = maxEncodedSize > MaxPasswordStackSize ?
+                (rentedPasswordBuffer = CryptoPool.Rent(maxEncodedSize)) :
+                stackalloc byte[MaxPasswordStackSize];
+            int passwordBytesWritten = s_throwingUtf8Encoding.GetBytes(password, passwordBuffer);
```

Comment:
This will throw an `EncoderFallbackException` when given an invalid string. Do we want to wrap it in `CryptographicException` or `ArgumentException`?

Reply:
In theory this should be `ArgumentException`, but my care level is minimal. We're not very consistent ourselves. :)

For example: _System.Formats.Asn1_ allows the EncoderFallbackException to bubble up. _System.Text.Json_ and _System.Formats.Cbor_ rewrap with ArgumentException (though _Cbor_ gives the wrong error text when doing so).

Reply:
Okay, my current inclination is to leave it as-is until someone more strongly suggests I change it.